### PR TITLE
Docker

### PR DESCRIPTION
**Pull request by MoisesTapia**
| Name   |                 About                   | Labels  | 
| ------- | ----------------------------- | -------  |
| Docker | Modify docker-compose.yml | suggestion-feature    |

# Sumary 
This PR is for it is to suggest changes in the docker-compose.yml since when we raise a docker-compose it creates an anonymous image

# Problem
- No container creation db
- No connection with DB
- No container creation ms
- No mode interactive with container ms error `stty: standard input: Not a tty`

## Running the Default docker-compose
- Enter the folder `metasploit-framework`
- Run th docker-compose with the command `docker-compose up`

## docker-compose
This is the docker-compose

your_env

## Pull and build images
In this images we can se how docker is pulling of **"db"** service and **"Building"** `metasploit:dev`

bulding_images

## Anonymous image

anonimate

## Image `metasploit:dev`

metasploiDEv

## Logs for docker-compose of **"db"** service

bd_name

## Logs for docker-compose of **"ms"**
ms_name
ms_exite

## `docker-compose ps`

docker_compose_ps

# Verification

- [ ] Change the name to `docker-compose.override.yml` to `.docker-compose.override.yml`
- [ ] Make a folder in home `mkdir .msf4` and change owner `sudo chown 1000`
- [ ] Return to folder `metasploit-framework`
- [ ] Write in terminal with logs `docker-compse up` withput logs or `docker-compose up -d`
- [ ] Enter the container with the command `docker-compose exec metasploit docker/entrypoint.sh ./msfconsole -r docker/msfconsole.rc -y $APP_HOME/config/database.yml`
- [ ] Verify with the command `docker-compose ps` that the container's state is UP
- [ ] Verify that when we are inside of metasploit's container this show us that the database is connected
- [ ] Verify that the metasploit's container doesn't show the output `stty: standard input: Not a tty` and instead shows us the output `stty: standard input`

### db pull

db_pull_done

### db up

db_up_done

### metasploit pull

metasploit_pull_done

### metasploit up 

metasploit_up_done

## Enter metasploit

metasploit_up_db_done

## `docker-compose ps`

dockerps_up_done

### CODE REVIEW

**Comment by MoisesTapia** [comment]
metasploitdocker
msf4

the docker-compose it's working

**Comment by gwillcox-r7** [comment]
Hello @MoisesTapia! Thanks for your first PR!  Unfortunately you didn't update the default issue template, so its very hard for us to tell what this PR is actually contributing. Would you mind please updating the PR description and using the template to describe what it is that you are submitting with this PR? This will help us process your contribution faster :)

**Comment by MoisesTapia** [comment]
@gwillcox-r7 of course but where i can find a template RP ?

**Comment by MoisesTapia** [comment]
@gwillcox-r7 Done ....

**Comment by MoisesTapia** [comment]
I have a question about my RP, is it correct and clear or not?

**Comment by gwillcox-r7** [comment]
@MoisesTapia Generally the normal format is as follows for reference:

1. Overview of the problem
2. Description of how to trigger the error
3. Description of what the PR fixes
4. Verification steps to show how to check that the PR actually fixes the error, with detailed step by step instructions. In your case I would like to see images of what it looks like before, and what steps to perform to fix the bug, and what output we should be seeing after. Please leave these check boxes unchecked, as they will be checked off by one of the Rapid7 members or contributors once they have verified that they were able to replicate your steps.

Please refer to some other PRs such as https://github.com/rapid7/metasploit-framework/pull/13733 for an idea of how one might provide the verification steps.

Overall the new info you posted is a lot better but I noticed a few screenshots such as the one under `Demo with logs` that don't have the command you executed to get that output. There is also still a little bit of confusion due to the lack of clear replication steps in some areas as your verification steps don't really explain how to check any of the issues your mentioned, all I have to go off to try replicate this issue is the screenshots which don't always provide the full context needed to replicate the original issue.

Looking better though, just needs a little more explanation in some places :)

**Comment by MoisesTapia** [comment]
@gwillcox-r7 Thanks and I will

File: docker-compose.yml [hunk 1]
```diff
@@ -1,21 +1,33 @@
 version: '3'
+
 services:
-  ms:
+
+  metasploit:
+    container_name: metasploit
```
Comment (adfoster-r7) [inline docker-compose.yml:6]: Hey, just a quick question on this line - is it possible to remove this `container_name` attribute and use `docker-compose exec` instead of `docker exec` to resolve containers?

```
docker-compose exec metasploit docker/entrypoint.sh ./msfconsole
```

Note that this would work in a similar way to the `docker exec` command that you've added a demo for.
Would that still work with your use case? 🤔

Comment (MoisesTapia) [reply docker-compose.yml:6]: @adamgalway-r7  Yes we can remove the `container name` attribute but the way of execute is diferent

Without `container_name`

dockerpserror

With `container_name`:

withcontainername

if we remove the `container_name` we have to access with the name or id of the container

**Comment by MoisesTapia** [comment]
I hope my documentation are complete @gwillcox-r7 ....

**Comment by gwillcox-r7** [comment]
```
- [ ] Build a docker container with docker-compose
- [ ] Fix the problem with the connection with containers
- [ ] Fix the problem with DB connection of metasploit and PostgreSQL
- [ ] Fix the problem with `stty: standard input: Not a tty`
```

@MoisesTapia Overall looking a lot better but these steps are not clear. For the first line, you haven't specified the full command to build the docker container, you just specified part of the command itself. If `docker-compose` is meant to be the command itself in full (aka you didn't pass it any parameters, files or other configuration options), then please wrap it in single quotes so that it shows up as code.

As for the last three items on that list, those are not verification steps. Those are requests. The verification steps are not meant to be for that. An example of a verification step would be `- [ ] Verify that the error "stty: standard input: Not a tty" no longer shows up when executing the command "blah blah blah"`. This is what I would expect to see: detailed instructions on what command to execute and what the change in the program's behavior should be once we apply your fixes.

Also for reference please in future drop your output as text rather than in images. You can do code listings by using 3 single quote characters on one line to mark the beginning of a block of text in Markdown, and end that block of text with another set of 3 single quote characters on one line.

So like so:

```
I'm test code
```

This will make it easier for people to copy and paste commands and check output if there are lots of steps to verify, and makes it easier for us to search for text within your error report should we need to double check certain output.

TLDR: Again, lot better than before, those lines above are only issue that I think definitely need fixing, the rest is more recommendations. I think @jmartin-r7 should be able to work off of this now, I was just here to ensure the original issue report is of a quality that @jmartin-r7 can easily pick it up :) Also edited your issue a bit to fix up some grammar errors as I noticed you did a few typos and had some odd wording. Nothing too major though :)

**Comment by MoisesTapia** [comment]
@gwillcox-r7 thanks for the feedback

**Comment by gwillcox-r7** [comment]
@MoisesTapia  Looks good, just making a few touch up fixes to fix some typos :)

**Comment by Oneiroi** [comment]
@gwillcox-r7 are we likely to see this merged soon-ish hit this bug myself last Friday, thanks :)

**Comment by jmartin-r7** [comment]
@Oneiroi, I am still reviewing this.  The approach is an improvement, however it also changes some behaviors.  Will bump this up in my queue though.

**Comment by Oneiroi** [comment]
@jmartin-r7 thank you :)

**Comment by gwillcox-r7** [comment]
@jmartin-r7 Any updates on testing this?

**Comment by gwillcox-r7** [comment]
@jmartin-r7 Checking in again, any updates on this?

**Comment by adfoster-r7** [comment]
@MoisesTapia Out of curiosity, have you used this script before? I think it allows you to achieve the same results

https://github.com/rapid7/metasploit-framework/blob/f132d353651acab5eb545244dc75f6c0f476de14/docker/bin/msfconsole

**Comment by gwillcox-r7** [comment]
@jmartin-r7 @MoisesTapia @adfoster-r7 No response received after 2 weeks and this has been left alone for over two years now with no updates. I'm putting this into the attic for now. Hopefully someone can fix this at some point but for the moment it doesn't seem like there is any ongoing work on this so the attic seems like a better place for now.

Feel free to unattic this when we have time as I feel there may still be issues like the one @Oneiroi mentioned here, however we have updated the Docker file a few times since this issue was raised so those issues may or may not still exist.

**Comment by ilcapone** [comment]
some news abaout this?!?

**Comment by gwillcox-r7** [comment]
> some news abaout this?!?

See above discussion, we don't presently have an update at this time.

**Comment by Oneiroi** [comment]
I'll retest once I have a cycle see if the issue persists, starting a new role today, so may be delayed in my response on this.
